packages/@aws-cdk/aws-batch/lib/compute-environment-base.ts

```diff
@@ -0,0 +1,119 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { IResource, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnComputeEnvironmentProps } from './batch.generated';
+
+
+/**
+ * Represents a ComputeEnvironment
+ */
+export interface IComputeEnvironment extends IResource {
+  /**
+   * The name of the ComputeEnvironment
+   *
+   * @default - generated by CloudFormation
+   *
+   * @attribute
+   */
+  readonly computeEnvironmentName?: string;
+
+  /**
+   * The ARN of this compute environment.
+   *
+   * @attribute
+   */
+  readonly computeEnvironmentArn: string;
+
+  /**
+   * The role Batch uses to perform actions on your behalf
+   * in your account, such as provision instances to run your jobs
+   *
+   * @default - a serviceRole will be created for managed CEs, none for unmanaged CEs
+   */
+  readonly serviceRole?: iam.IRole;
+
+  /**
+   * Whether or not this ComputeEnvironment can accept jobs from a Queue.
+   * Enabled ComputeEnvironments can accept jobs from a Queue and
+   * can scale instances up or down.
+   * Disabled ComputeEnvironments cannot accept jobs from a Queue or
+   * scale instances up or down.
+   *
+   * If you change a ComputeEnvironment from enabled to disabled while it is executing jobs,
+   * Jobs in the `STARTED` or `RUNNING` states will not
+   * be interrupted. The ComputeEnvironment will no longer scale instances down;
+   * that is, any instances that were scaled up will remain if the ComputeEnvironment is disabled.
+   *
+   * To ensure you aren't billed for unused capacity, remove the ComputeEnvironment from your stack.
+   */
+  readonly enabled: boolean;
```

Comment:
We set it to `true` and give empty values where needed:

```
  public static fromManagedEc2EcsComputeEnvironmentArn(
    scope: Construct, id: string, managedEc2EcsComputeEnvironmentArn: string,
  ): IManagedEc2EcsComputeEnvironment {
    const stack = Stack.of(scope);
    const computeEnvironmentName = stack.splitArn(managedEc2EcsComputeEnvironmentArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName!;

    class Import extends Resource implements IManagedEc2EcsComputeEnvironment {
      public readonly computeEnvironmentArn = managedEc2EcsComputeEnvironmentArn;
      public readonly computeEnvironmentName = computeEnvironmentName;
      public readonly enabled = true;
      public readonly instanceClasses = [];
      public readonly instanceTypes = [];
      public readonly maxvCpus = 1;
      public readonly connections = { } as any;

      public addInstanceClass(_instanceClass: ec2.InstanceClass): void {
        throw new Error(`cannot add instance class to imported ComputeEnvironment '${id}'`);
      }
      public addInstanceType(_instanceType: ec2.InstanceType): void {
        throw new Error(`cannot add instance type to imported ComputeEnvironment '${id}'`);
      }
    }

    return new Import(scope, id);
  }
```

Reply:
I understand, but who *needs* it? What other construct cannot operate without this exposed?

Because it seems to be designed for use by subclasses, but the subclass initialization is awkward currently.